### PR TITLE
[WIP] feat: implementation of wishlist

### LINKED ISSUE #4988: Implement whishlist feature
## What is the motivation for adding / enhancing this feature?
As commercetools has introduced new queries for managing wishlist we are able to cover this feature now.

## What are the acceptance criteria 
* create api-client function `createMyShopingList` that uses mutation `createMyShoppingList`
* create api-client function  `updateMyShoppingList` that uses mutation `updateMyShoppingList`
* create api-client function `getMyShoppingList ` that uses user query (look at getMe) but limited just for fields related to the whiwhlist
* implement composable according to type UseWhishlist with usingUseWhishlsitFactory
* implement getters according to type UseWhishlistGetters

## Version of Vue Storefront
* [ ] Vue Storefront
* [x] Vue Storefront Next

## Can you complete this feature request by yourself?
* [x] YES
* [ ] NO

## Additional information
When it comes commercetools wishlist is settled inside of user (that's why you have to use user query). Because of that, you actually will call the user once again, but for the different fields (power of gql)

┆Issue is synchronized with this [Jira Story](https://vsf.atlassian.net/browse/NEXT-327) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by AdamPawlinski**
### Related Issues
closes #4988 

### Short Description and Why It's Useful
[WIP]
Implementation of API-client functions to create, update and get shopping list and add composable.

### Which Environment This Relates To
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [ ] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog
- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [ ] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/docs/guide/upgrade-notes/README.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing Vue Storefront sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: packages/commercetools/api-client/src/api/createMyShoppingList/defaultMutation.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import gql from 'graphql-tag';
+
+export default gql`
+  mutation createMyShoppingList($draft: MyShoppingListDraft!, $locale: Locale!, $acceptLanguage: [Locale!], $storeKey: KeyReferenceInput, $currency: Currency!) {
+    shoppingList: createMyShoppingList(draft: $draft, storeKey: $storeKey, locale: $locale, currency: $currency)
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/createMyShoppingList/defaultMutation.ts:5]: whishlist

File: packages/commercetools/api-client/src/api/createMyShoppingList/defaultMutation.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      slug
+      lineItems {
+        id 
+        productId
+        variantId
+        quantity
+        variant {
+          id
+          sku
+          price (currency: "USD") {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/createMyShoppingList/defaultMutation.ts:18]: currency: $currency

File: packages/commercetools/api-client/src/api/createMyShoppingList/index.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { apolloClient, getSettings } from './../../index';
+import defaultQuery from './defaultMutation';
+import { getCustomQuery } from './../../helpers/queries';
+import gql from 'graphql-tag';
+
+interface ShoppingListData extends Omit<MyShoppingListDraft, 'currency'> {
+  currency?: string;
+}
+
+const createMyShoppingList = async (myShoppingListDraft: ShoppingListData, customQueryFn?) => {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/createMyShoppingList/index.ts:12]: draft

File: packages/commercetools/api-client/src/api/createMyShoppingList/index.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  const { locale, acceptLanguage, currency } = getSettings();
+
+  const defaultVariables = {
+    draft: {
+      acceptLanguage,
+      locale,
+      currency,
+      ...myShoppingListDraft
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/createMyShoppingList/index.ts:20]: draft

File: packages/commercetools/api-client/src/api/getMyShoppingList/defaultQuery.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        slug
+        lineItems {
+          id 
+          productId
+          variantId
+          quantity
+          variant {
+            id
+            sku
+            price(currency: "USD") {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/getMyShoppingList/defaultQuery.ts:19]: $currency

File: packages/commercetools/api-client/src/api/getMyShoppingList/defaultQuery.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
+import gql from 'graphql-tag';
+
+const basicProfile = gql`
+  query getMe($locale: Locale!, $acceptLanguage: [Locale!]) {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/getMyShoppingList/defaultQuery.ts:4]: $currency

File: packages/commercetools/api-client/src/api/updateMyShoppingList/defaultMutation.ts [hunk 7]
```diff
@@ -0,0 +1,50 @@
+import gql from 'graphql-tag';
+
+export default gql`
+
+  mutation updateShoppingList($id: String!, $version: Long!, $actions: [MyShoppingListUpdateAction!]!, $locale: Locale!, $acceptLanguage: [Locale!]) {
+    shoppingList: updateMyShoppingList(id: $id, version: $version, actions: $actions) {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/updateMyShoppingList/defaultMutation.ts:6]: whishlist

File: packages/commercetools/api-client/src/api/updateMyShoppingList/defaultMutation.ts [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      slug
+      lineItems {
+        id 
+        productId
+        variantId
+        quantity
+        variant {
+          id
+          sku
+          price (currency: "USD") {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/updateMyShoppingList/defaultMutation.ts:19]: $currency

File: packages/commercetools/api-client/src/api/updateMyShoppingList/index.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+import gql from 'graphql-tag';
+import { getCustomQuery } from './../../helpers/queries';
+
+interface UpdateMyShoppingList {
+  id: string;
+  version: number;
+  actions: MyShoppingListUpdateAction[];
+}
+
+const updateMyShoppingList = async (params: UpdateMyShoppingList, customQueryFn?) => {
```
Comment (andrzejewsky) [inline packages/commercetools/api-client/src/api/updateMyShoppingList/index.ts:13]: please use the product list as arguments

File: packages/commercetools/composables/src/useWishlist/currentWishlist.ts [hunk 10]
```diff
@@ -0,0 +1,16 @@
+import { createMyShoppingList, getMe } from '@vue-storefront/commercetools-api';
+
+const loadCurrentShoppingList = async (customQueryFn = (user = null, wishlist = null) => ({ user, wishlist })) => {
+  const { user, wishlist } = customQueryFn();
+  const { data: profileData } = await getMe({ customer: false }, user);
+
+  if (profileData.me.shoppingList) {
+    return profileData.me.shoppingList;
```
Comment (andrzejewsky) [inline packages/commercetools/composables/src/useWishlist/currentWishlist.ts:8]: whishlist

**Review by andrzejewsky — CHANGES_REQUESTED** [review]

File: packages/commercetools/api-client/src/api/createMyShoppingList/index.ts (hunk 3 above)
Comment (AdamPawlinski) [reply packages/commercetools/api-client/src/api/createMyShoppingList/index.ts:12]: @andrzejewsky Can you remind me please what was the problem here?

**Comment by AdamPawlinski** [comment]
Wishlist issue is continued on enterprise repo.

**Pull request closed without merging**